sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -266,7 +301,10 @@ cdef class Splitter:
                 self.X_binned[:, feature_idx]
             unsigned int [::1] left_indices_buffer = self.left_indices_buffer
             unsigned int [::1] right_indices_buffer = self.right_indices_buffer
-
+            unsigned char is_categorical = split_info.is_categorical
+            # we need a tmp variable because left_cat_bitset may be None
+            BITSET_INNER_DTYPE_C [:] cat_bitset_tmp = split_info.left_cat_bitset
+            BITSET_DTYPE_C left_cat_bitset
```

Comment:
No

Reply:
Is the temporary then really necessary? Wen we later assign
```
if is_categorical:
            left_cat_bitset = &cat_bitset_tmp[0]
```

Reply:
I agree it's unsettling, I tried to remove it a while ago, and couldn't find something simple. I don't remember what it was, but everything I tried made Cython yell at me :/

Reply:
I also find no other solution. We could, however, move the initialization down, i.e.
```
cdef:
    BITSET_INNER_DTYPE_C [:] cat_bitset_tmp

...

if is_categorical:
    cat_bitset_tmp = split_info.left_cat_bitset
    left_cat_bitset = &cat_bitset_tmp[0]
```

Reply:
How about the suggested later initialization of the memoryview?

Reply:
Can you help me understand the benefit of initializing later?

We still have to define the tmp variable. And what it saves in term of speed is just assigning the view to `None`.

Reply:
Nothing more, I guess.
What bothers me a bit is the comment `# we need a tmp variable because left_cat_bitset may be None` which I find misleading. More honest would be that we don't find another way to make Cython happily assign the pointer of `split_info.left_cat_bitset` to `left_cat_bitset`.